validation.go

```diff
@@ -53,7 +73,21 @@ func (v Validator) VerifyRecord(r *pb.Record) error {
 		return ErrInvalidRecordType
 	}
 
-	return val.Func(r.GetKey(), r.GetValue())
+	var author *peer.ID
+	if len(r.GetSignature()) > 0 {
+		pid, err := peer.IDFromString(r.GetAuthor())
+		if err != nil {
+fmt.Printf("Could not parse author to peer ID: %s\n", r.GetAuthor())
+			return ErrInvalidRecordType
+		}
+		author = &pid
+	}
+	vr := &ValidationRecord{
```

Comment:
so how are we going to validate the signature if we don't pass it?

Reply:
Or is this something we do at a different layer?

Reply:
This is where the DHT checks it:
https://github.com/libp2p/go-libp2p-kad-dht/blob/5a05448e7cf4003ea56a864c40fb371195adf6e1/records.go#L129

Reply:
ok, so the signature is considered validated at this layer.